crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -595,7 +606,13 @@ impl Timeline {
             }
         }
 
-        self.room().send_multiple_receipts(receipts).await
+        if !receipts.is_empty() {
+            self.room().send_multiple_receipts(receipts).await?;
+        } else if self.room().is_marked_unread() {
+            self.room().set_unread_flag(false).await?;
```

Comment:
Instead of specializing each call site of `Room::set_unread_flag()`, consider making `Room::set_unread_flag` a no-op if `!self.is_marked_unread()`?